src/Symfony/Component/Routing/RouteCollection.php

```diff
@@ -157,15 +157,8 @@ public function get(string $name)
      */
     public function remove($name)
     {
-        $names = (array) $name;
-        foreach ($names as $n) {
-            unset($this->routes[$n], $this->priorities[$n]);
-        }
-
-        foreach ($this->aliases as $k => $alias) {
-            if (\in_array($alias->getId(), $names, true)) {
-                unset($this->aliases[$k]);
-            }
+        foreach ((array) $name as $n) {
+            unset($this->routes[$n], $this->priorities[$n], $this->aliases[$n]);
```

Comment:
Maybe I'm missing something here, but why not have this method remove both an alias, if it's an alias, or a route with all its aliases, if it's a route?

Reply:
I also wondered the same. That'd make sense to me.